client/checkout/classic/new-upe.js

```diff
@@ -0,0 +1,242 @@
+	PAYMENT_METHOD_NAME_IDEAL,
+	PAYMENT_METHOD_NAME_P24,
+	PAYMENT_METHOD_NAME_SEPA,
+	PAYMENT_METHOD_NAME_SOFORT,
+} from '../constants.js';
+import {
+	getSelectedUPEGatewayPaymentMethod,
+	getUpeSettings,
+} from '../utils/upe';
+import showErrorCheckout from '../utils/show-error-checkout';
+import { getAppearance } from '../upe-styles';
+import {
+	appendFingerprintInputToForm,
+	getFingerprint,
+} from '../utils/fingerprint';
+
+jQuery( function ( $ ) {
+	const paymentMethods = getUPEConfig( 'paymentMethodsConfig' );
+	const gatewayUPEComponents = {};
+	const api = new WCPayAPI(
+		{
+			publishableKey: getUPEConfig( 'publishableKey' ),
+			accountId: getUPEConfig( 'accountId' ),
+			forceNetworkSavedCards: getUPEConfig( 'forceNetworkSavedCards' ),
+			locale: getUPEConfig( 'locale' ),
+		},
+		apiRequest
+	);
+
+	let fingerprint = null;
+	let appearance = getUPEConfig( 'upeAppearance' );
+
+	enqueueFraudScripts( getUPEConfig( 'fraudServices' ) );
+	initAppearance();
+	initStripeElements();
+
+	function initAppearance() {
+		if ( ! appearance ) {
+			appearance = getAppearance();
+			api.saveUPEAppearance( appearance );
+		}
+	}
+
+	function initStripeElements() {
+		for ( const paymentMethodType in paymentMethods ) {
+			gatewayUPEComponents[ paymentMethodType ] = {
+				elements: null,
+				upeElement: null,
+			};
+		}
+	}
+
+	/**
+	 * This is the entry-point of this file, which is responsible for mounting the UPE elements.
+	 * It is called when the page is loaded.
+	 *
+	 */
+	$( document.body ).on( 'updated_checkout', () => {
+		$( '.wcpay-upe-element' )
+			.toArray()
+			.forEach( ( domElement ) => {
+				const paymentMethodType = $( domElement ).data(
+					'payment-method-type'
+				);
+				const upeElement =
+					gatewayUPEComponents[ paymentMethodType ].upeElement;
+				if ( upeElement ) {
+					upeElement.mount( domElement );
+				} else {
+					createAndMountPaymentElement(
+						paymentMethodType,
+						domElement
+					);
+				}
+			} );
+	} );
+
+	/**
+	 * Block UI to indicate processing and avoid duplicate submission.
+	 *
+	 * @param {Object} $form The jQuery object for the form.
+	 */
+	const blockUI = ( $form ) => {
+		$form.addClass( 'processing' ).block( {
+			message: null,
+			overlayCSS: {
+				background: '#fff',
+				opacity: 0.6,
+			},
+		} );
+	};
+
+	/**
+	 * Creates the UPE element and mounts it to the DOM element.
+	 * No Stripe PaymentIntent is created at this point.
+	 *
+	 * @param {string} paymentMethodType The type of payment method, e.g. 'card', 'giropay', etc.
+	 * @param {string} domElement The selector of the DOM element of particular payment method to mount the UPE element to.
+	 */
+	async function createAndMountPaymentElement(
+		paymentMethodType,
+		domElement
+	) {
+		const amount = Number( getUPEConfig( 'cartTotal' ) );
+		const options = {
+			mode: 1 > amount ? 'setup' : 'payment',
+			currency: getUPEConfig( 'currency' ).toLowerCase(),
+			amount: Number( getUPEConfig( 'cartTotal' ) ),
```

Comment:
```suggestion
			amount: amount,
```
Could just reuse the var you defined above, no?

Reply:
Sure, fixed in 40031cf9a4c6c420589b640bde7eba2745070546